src/main/java/com/minecolonies/coremod/network/messages/BuildToolPlaceMessage.java

```diff
@@ -124,36 +125,39 @@ public void toBytes(@NotNull final ByteBuf buf)
     public void messageOnServerThread(final BuildToolPlaceMessage message, final EntityPlayerMP player)
     {
         final World world = player.world;
+        final Structures.StructureName sn = new Structures.StructureName(message.structureName);
+        if (!Structures.hasMD5(sn))
+        {
+            player.sendMessage(new TextComponentString("Can not build " + message.workOrderName + ": schematic missing!"));
+            return;
+        }
         if (message.isHut)
         {
-            handleHut(world, player, message.hutDec, message.style, message.rotation, message.pos, message.mirror);
+            handleHut(world, player, sn, message.workOrderName, message.rotation, message.pos, message.mirror);
         }
         else
         {
-            handleDecoration(world, player, message.hutDec, message.style, message.rotation, message.pos, message.mirror);
+            handleDecoration(world, player, sn, message.workOrderName, message.rotation, message.pos, message.mirror);
         }
     }
 
     /**
      * Handles the placement of huts.
      *
-     * @param world    World the hut is being placed into.
-     * @param player   Who placed the hut.
-     * @param hut      The hut we are placing.
-     * @param style    The style of the hut.
-     * @param rotation The number of times the structure should be rotated.
-     * @param buildPos The location the hut is being placed.
+     * @param world         World the hut is being placed into.
+     * @param player        Who placed the hut.
+     * @param sn            The name of the structure.
+     * @param workOrderName The name of the work order.
+     * @param rotation      The number of times the structure should be rotated.
+     * @param buildPos      The location the hut is being placed.
+     * @param mirror        Whether or not the strcture is mirrored.
      */
     private static void handleHut(
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 26 to the 20 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)